vllm/v1/engine/detokenizer.py

```diff
@@ -125,7 +126,7 @@ def update(self, new_token_ids: list[int],
 
         # 2) Evaluate stop strings.
         stop_string = None
-        if self.stop:
+        if self.stop and len(self.output_token_ids) >= self.min_tokens:
```

Comment:
There's technically an edge case with spec decoding where multiple tokens are produced in this step, and the stop sequence occurs in the decoded text before the minimum token count is reached. But I'm not sure that it's worth the additional complexity to handle this.